Plotly.Blazor/PlotlyJsInterop.cs

```diff
@@ -1,348 +1,367 @@
+    public async Task SubscribeHoverEvent(CancellationToken cancellationToken)
+    {
+        var jsRuntime = await moduleTask.Value;
+
+        await jsRuntime.InvokeVoidAsync("subscribeHoverEvent", cancellationToken, dotNetObj, dotNetObj.Value.Id);
+    }
+
+    /// <summary>
+    ///     Can be used to subscribe selected events for points.
+    /// </summary>
+    /// <param name="cancellationToken">CancellationToken</param>
+    public async Task SubscribeSelectedEvent(CancellationToken cancellationToken)
+    {
+        var jsRuntime = await moduleTask.Value;
+
+        await jsRuntime.InvokeVoidAsync("subscribeSelectedEvent", cancellationToken, dotNetObj, dotNetObj.Value.Id);
+    }
+
+    /// <summary>
+    ///     Can be used to subscribe to relayout events.
+    /// </summary>
+    /// <param name="cancellationToken"></param>
+    public async Task SubscribeRelayoutEvent(CancellationToken cancellationToken)
+    {
+        var jsRuntime = await moduleTask.Value;
+
+        await jsRuntime.InvokeVoidAsync("subscribeRelayoutEvent", cancellationToken, dotNetObj, dotNetObj.Value.Id);
+    }
+
+    /// <summary>
+    ///     Can be used to subscribe to restyle events.
+    /// </summary>
+    /// <param name="cancellationToken"></param>
+    public async Task SubscribeRestyleEvent(CancellationToken cancellationToken)
+    {
+        var jsRuntime = await moduleTask.Value;
+
+        await jsRuntime.InvokeVoidAsync("subscribeRestyleEvent", cancellationToken, dotNetObj, dotNetObj.Value.Id);
+    }
+
+    /// <summary>
+    ///     Can be used to export the chart as a static image and returns a binary string of the exported image.
+    /// </summary>
+    /// <param name="format">Format of the image.</param>
+    /// <param name="height">Height of the image.</param>
+    /// <param name="width">Width of the image.</param>
+    /// <returns>Binary string of the exported image.</returns>
+    /// <param name="cancellationToken">CancellationToken</param>
+    public async Task<string> ToImage(ImageFormat format, uint height, uint width, CancellationToken cancellationToken)
+    {
+        var jsRuntime = await moduleTask.Value;
+
+        return await jsRuntime.InvokeAsync<string>("toImage", cancellationToken, dotNetObj.Value.Id, format, height, width);
+    }
+
+    /// <summary>
+    ///     Can be used to export the chart as a static image and returns a binary string of the exported image.
+    /// </summary>
+    /// <param name="chartData">The chart data to be exported.</param>
+    /// <param name="format">Format of the image.</param>
+    /// <param name="height">Height of the image.</param>
+    /// <param name="width">Width of the image.</param>
+    /// <param name="cancellationToken">CancellationToken</param>
+    /// <returns>Binary string of the exported image.</returns>
+    public async Task<string> ToImage(object chartData, ImageFormat format, uint height, uint width, CancellationToken cancellationToken)

```

Comment:
Can you maybe use the same logic like in `Relayout<T>` for serialization of an anonymous type.

Reply:
So people dont need "dynamics" :)